src/client/components/loginSignUpPage/SignUpPage.jsx

```diff
@@ -1,76 +1,124 @@
 import React, { useState } from 'react';
-import { Button, Icon, Form } from 'react-bulma-components';
 import styled from 'styled-components';
-import { OrangeButton } from '/src/client/styles/shared.tsx';
 import axios from 'axios';
+import { Button, Icon, Form } from 'react-bulma-components';
 import { useHistory } from 'react-router-dom'
 import { OrangeNavbar } from '../../styles/shared';
+import { OrangeButton } from '/src/client/styles/shared.tsx';
+import validate from './validationRules.js';
+import useForm from './formValidation.js';
 
 const StyledHeader = styled.h2`
 font-size: 20px;
 font-weight: bold;
+margin-bottom: 5px;
 `;
 
 export default function SignUpPage() {
-  const [firstName, setFirstName] = useState('');
-  const [lastName, setLastName] = useState('');
-  const [email, setEmail] = useState('');
-  const [password, setPassword] = useState('');
-
-  const handleSignUp = () => {
-    axios.post('', {
-      //object to send based on database structure
-    })
-    .then(res => {
-      console.log('successfully registered user');
-      //redirect user to the login page?
-    })
-    .catch(err => {
-      console.log(err)
-    })
-  }
+  const [username, setUsername] = useState('');
 
   const history = useHistory();
   function handleHomeClick() {
     history.push("/");
   }
+
   const goBack = () => {
     history.goBack()
   }
 
+  const signUp = () => {
+    axios.post('/create', {
+      first_name: values.firstName,
+      last_name: values.lastName,
+      email: values.email,
+      username: username,
+      password: values.password,
+      guest: false
+    })
+      .then(res => {
+        console.log('successfully registered user');
+        goBack();
+      })
+      .catch(err => {
+        console.log(err);
+      })
+  }
+
+  const { values, handleChange, errors, handleSubmit } = useForm(signUp, validate);
+
   return (
     <div className="login-signup-background">
-      < OrangeNavbar needBackArrow={true} onBackArrowClick={goBack}/>
+      < OrangeNavbar needBackArrow={true} onBackArrowClick={goBack} />
       <div className="login-signup-page-container">
         <StyledHeader>New Account</StyledHeader>
         <Form.Field className="login-form">
           <Form.Label>First Name</Form.Label>
-          <Form.Control>
-            <Form.Input placeholder="First Name" onChange={e => {setFirstName(e.target.value)}}/>
+          <Form.Control className="form-spacing-signup">
```

Comment:
I see why you're adding CSS styles into a stylesheet and not into a styled component. 

with styled components, to refer to a specific className for said component you can do something like this

```
const FormControl = styled(Form.Control)'
  &.login-signup-page-container {
    margin: 20%;
    display: flex;
    flex-direction: column;
    flex-wrap: wrap;
    align-items: center;
    justify-content: center;
   }'
```

the & refers to the component itself, so `&.className` is saying any of this styled component that i gave the classname of "className" to, i want this specific style. if you do just `.className` inside the styled component, it refers to any CHILD component that has the given classname

also just an fyi the above code is using single quotes not backticks like you are supposed to, so you gotta change it to a backtick!

https://styled-components.com/docs/basics#getting-started

Reply:
oo interesting didn't know about this! Thanks Erik!